File: PalindromeNumber/palindromeNumber.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let left = 0;
    let right = (x.toString().length) - 1;
    let answer = true;
    let arr_x = [...x.toString()];

    while (left <= right) {
      left += 1;
      right -= 1;
      if (arr_x[left] != arr_x[right]) {
        answer = false;
        break;
      }
    }
    return answer;
};


data = [121, -121, 10, 1234321]
data.forEach(element => {
  console.log(isPalindrome(element))
});

File: PalindromeNumber/palindromeNumber.py

class Solution:
  def isPalindrome(self, x: int) -> bool:
    stringifyN = str(x)
    left = 0
    right = len(stringifyN) - 1
    while left < right:
      if stringifyN[left] != stringifyN[right]:
        return False
      left += 1
      right -= 1
    return True


n = 43234
sol = Solution()
print(sol.isPalindrome(n))